src/jit/lir.cpp

```diff
@@ -528,9 +528,10 @@ GenTree* LIR::Range::FirstNonPhiOrCatchArgNode() const
 
     for (GenTree* node = FirstNonPhiNode(), *end = End(); node != end; node = node->gtNext)
     {
-        if ((node->OperGet() == GT_STORE_LCL_VAR) && (node->gtOp.gtOp1->OperGet() == GT_CATCH_ARG))
+        if ((node->OperGet() != GT_CATCH_ARG) &&
+            ((node->OperGet() != GT_STORE_LCL_VAR) || (node->gtOp.gtOp1->OperGet() != GT_CATCH_ARG)))
         {
```

Comment:
This logic is a bit hard to understand, IMO. What do you think about this instead:
```
if (node->OperGet() == GT_CATCH_ARG)
    continue;

if ((node->OperGet() == GT_STORE_LCL_VAR) && (node->gtOp.gtOp1->OperGet() == GT_CATCH_ARG))
    continue;

break;
```
and then the only "return" would be "return node" (which is correct for either loop exit).